### DISCUSSION TITLE
How do you _actually_ restrict imports (which `no-restricted-imports` doesn't seem to do?)

### DISCUSSION THREAD

**Discussion by matthew-dean** [eslint/eslint#18243] [Q&A]
So, I'm noticing that `no-restricted-imports` only prevents imports if how they resolve _matches_ one of the patterns given. But that may not always be the case.

For example, say you have a folder `components` and all components are exported by `index.js`. Then you add an eslint rule for all components within components to say that they must not import other components from index.js, so that circular dependencies are not created.

However, if you naively set `~/components` as a restricted import (with `~` an alias) and someone imports not from `~/components` but imports from `..`, which _resolves_ to `index.js`, then this rule will fail.

Is there a way to restrict imports by resolved path and not by the way people write an import?

**Comment by nzakas** [comment]
The rule does not resolve relative paths, so you'll need to explicitly match the patterns you want to block. You may want to block the use of `..` to make things easier.
